src/main.cpp

```diff
@@ -2630,19 +2630,19 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
             for (unsigned int k = 0; k < tx.vout.size(); k++) {
                 const CTxOut &out = tx.vout[k];
                 int const scriptType = out.scriptPubKey.Type();
-                if (!scriptType)
-                    continue;
-                uint160 const addrHash = uint160(out.scriptPubKey.AddressHash());
+                if (scriptType > 0) {
+                    uint160 const addrHash = uint160(out.scriptPubKey.AddressHash(scriptType));
```

Comment:
Missing check?  What happens when uint160 is called with an empty vector?  AddressHash() has been updated in the commit where it can return an empty vector.  If empty, the script type is invalid for the purposes of the explorer, and we should 'continue' to next iteration of loop.  (This comment applies everywhere AddressHash(scriptType) is used.)

Reply:
I think it's okay because `AddressHash()` only returns an empty vector if `scriptType` is zero, see https://github.com/zcash/zcash/pull/3837/files#diff-f7ca24fb80ddba0f291cb66344ca6fcbR256 but please double-check that for me.

Yes, I used to have exactly that, `if (scriptType == 0) continue` but changed it to this because later, when we add `spentindex` (#3863), we are going to want to do some stuff in the body of this loop even if `scriptType` is zero. So writing it this way makes adding that change easier.

Reply:
@LarryRuane So `AddressHash(scriptType)` is only invoked when `scriptType > 0`, but if `scriptType >= 3`, an empty vector is returned. Given that addressIndex only supports scriptType 1 or 2, maybe the conditional `if (scriptType > 0)` should be changed to reflect this e.g. `if (scriptType == 1 || scriptType == 2) { ... }`

Reply:
@bitcartel I think this code is correct as it is. We may recognize more script types in the future, say `scriptType == 3`, and this code won't have to change, and it shouldn't have to change. The convention is: If `Type()` returns non-zero, that means `AddressHash()`, when passed that return value, returns a legitimate address (not an empty vector). As long as the two methods, `Type()` and `AddressHash`, are kept in sync (and they're implemented right next to each other in `src/script/script.cpp`), then it will be okay. I think it will be pretty clear to future developers that those functions need to stay in sync. (But if you disagree, I could add a comment.)